src/shared/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { forwardRef, useState } from 'react'
+import NumberFormat from 'react-number-format'
+import { parse, isValid } from 'date-fns'
+import TextField from '../TextField'
+
+export type DatepickerProps = {
+  required?: boolean
```

Comment:
This component currently isn't really usable - there's no way to access its value from outside - the consumer needs to be able to read that date somehow. I guess we could add some `onChange` that would be called as part of the `onBlur` handler. Possibly the validation would be done outside as well. `onChange` would return either a valid or an invalid date, and the form validation would set the error based on that?